webservice/src/main/java/com/hortonworks/streamline/webservice/StreamlineApplication.java

```diff
@@ -217,11 +214,14 @@ private CacheBuilder getGuavaCacheBuilder() {
         return CacheBuilder.newBuilder().maximumSize(maxSize);
     }
 
-    private FileStorage getJarStorage (StreamlineConfiguration configuration) {
+    private FileStorage getJarStorage (StreamlineConfiguration configuration, StorageManager storageManager) {
         FileStorage fileStorage = null;
         try {
             fileStorage = ReflectionHelper.newInstance(configuration.getFileStorageConfiguration().getClassName());
             fileStorage.init(configuration.getFileStorageConfiguration().getProperties());
+            if (fileStorage instanceof DbFileStorage) {
```

Comment:
If we're a fan of interface, we even could have an interface `StorageManagerAware` and let DbFileStorage implement it. Maybe more flexible given that FileStorage is a kind of plug-in.